src/NLog/SetupBuilderExtensions.cs

```diff
@@ -106,6 +114,15 @@ public static ISetupBuilder LoadConfiguration(this ISetupBuilder setupBuilder, A
             {
                 setupBuilder.LogFactory.Configuration = newConfig;
             }
+#if !NETSTANDARD1_3
+            if (applyOnReload)
+            {
+                setupBuilder.LogFactory.ConfigurationChanged += (sender, e) =>
```

Comment:
ConfigurationChanged will give notifications on the following events:
- AutoReload - FileWatcher-logic
- ManualLoad - Assigning `LogFactory.Configuration = newConfig;`
- ManualUnload - Assigning `LogFactory.Configuration = null;`
- Shutdown

I think you should not perform the callback when there is no active LoggingConfiguration available. And maybe consider if the callback should also happen on `ManualLoad`.

Reply:
I'm also doubting with ManualLoad , but that one is hard to prevent now isn't?

Reply:
> with ManualLoad , but that one is hard to prevent now isn't?

I guess for XmlLoggingConfiguration then one could check if config was assigned using LogFactory.Configuration-property-setter (And signal that through the ConfigurationChanged-EventArgs). But for NLogLoggingConfiguration then it is not easy, since it has to use the official setter-gateway.

So yes hard to distinguish AutoReload and ManulLoad.

Reply:
Good to know is that KeepVariablesOnReload is also working for manual load.

I have some things to fix here, but we should have a clear direction on this. If manual load is a no-go, I can't fix this (or not sure how)